### PR TITLE
Adding changelog for v1.0.0-rc1

### PR DESCRIPTION
**Pull request by robscott**
**What type of PR is this?**
/kind documentation

**What this PR does / why we need it**:
This adds the changelog for v1.0.0-rc1. By approving this PR, you're also approving the release of v1.0.0-rc1.

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

/cc @thockin @khenidak @MikeZappa87
/assign @bowei @shaneutt @youngnick 
/hold for consensus

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -31,10 +32,107 @@
...
+
+This release candidate is suitable for implementors, but the working group does
+not recommend shipping products based on a release candidate API due to the
+possibility of incompatible changes prior to the final release. The following
+represents the changes since v0.8.0-rc1:
+
+## Gateway, GatewayClass, and HTTPRoute are GA 🎉
+
+Gateway, GatewayClass, and HTTPRoute have all graduated to GA with a `v1` API
+version. Although these APIs will continue to grow with future additions, the
```
Comment (shaneutt) [inline CHANGELOG.md:48]: </img>

File: CHANGELOG.md [hunk 2]
```diff
@@ -49,6 +147,28 @@
...
+  Standard and Experimental. New resources will start in the Experimental
+  Channel with an alpha API Version and then graduate to the Standard Channel
+  with a GA API version. Resources that already have Beta API versions will
+  continue to have them, but no additional Gateway API resources will get a Beta
+  API version. (#2446, @robscott)
+
+### Cleanup
+- Validating Webhook logs now use `Errorf` instead of `Fatalf`. (#2361, @yylt)
+
+### Other (Cleanup or Flake)
```
Comment (shaneutt) [inline CHANGELOG.md:163]: not a blocker persay, but should we draw extra attention to this item?

**Review by shaneutt — APPROVED** [review]

File: CHANGELOG.md (hunk 2 above)
Comment (robscott) [reply CHANGELOG.md:179]: Yep, good call, added new section above, thanks!

**Pull request merged**